### ISSUE TITLE
Deploying to Github Pages without a custom domain name

### ISSUE THREAD

**Issue by katerberg** [tweetback/tweetback#45]
After following the instructions [here](https://github.com/tweetback/tweetback/blob/main/docs/deploy-with-github-pages.md), I had a site deployed on Github Pages, but all the CSS was broken only on Github Pages (locally worked fine):

<img width="940" alt="Screen Shot 2022-12-04 at 9 31 12 PM" src="https://user-images.githubusercontent.com/479643/205545578-44fda001-db6d-44b7-8f72-fa9e424b02e5.png">

I noticed in the devconsole a cryptic error: `Uncaught ReferenceError: getSentimentsFromList is not defined`. And then saw that `chartist.min.js` and many other imports were broken because they were referencing `https://katerberg.github.io/assets/chartist.min.js` instead of `https://katerberg.github.io/tweetback/assets/chartist.min.js`. 

I tried using `pathPrefix` in the `eleventy.config.js` but it did not change the output after running `npm run build` and neither did passing it via a command line option.

In the end I got it all working by updating all of the imports to include the subdomain matching my repository: `tweetback` via the following steps:

- Added `eleventyConfig.pathPrefix = "/tweetback/";` to the bottom of the config.
- Added `eleventyConfig.addPassthroughCopy({ "assets/": "tweetback/assets/" });` along with the other passthroughs
- Updated passthroughs for `chartist` et al to point to the `/tweetback` prefixed value
- Updated `_includes/layout.11ty.js` to have anything that started with `/assets` to instead start with `/tweetback/assets`

I think this would probably not be an issue if people are hosting it with a custom domain and not using a subdomain for the hosting, but it might be worth updating the docs to indicate this.

See the final working version here: https://github.com/katerberg/tweetback

**Comment by jakejarvis** [comment]
@katerberg Thanks for this! For anyone else stumbling upon this like I did:

These steps worked perfectly, except for 2 and 3 which appear unnecessary — the `addPassthroughCopy` changes caused these files to output to `_site/tweetback/tweetback/assets`, which means the absolute URLs become something like `you.github.io/tweetback/tweetback/assets/chart.js`.

So the changes for this seem as simple as:
1. adding `eleventyConfig.pathPrefix = "/tweetback/";` to the bottom of `module.exports` in `next.config.js`
2. doing a find + replace for `/assets` ➡️ `/tweetback/assets` in `_includes/layout.11ty.js`

And if your GH pages repo is named something other than `tweetback`, use that instead for both steps (e.g. `/assets/twitter`).

Will try to word all of this less confusingly for a PR when I have a chance, unless someone beats me to it!

🎉 https://jakejarvis.github.io/tweets/

**Comment by jakejarvis** [comment]
Update: bumping `@11ty/eleventy` to `2.0.0-canary.20` indeed made my mess above entirely unnecessary! 

It's now as simple as adding a few lines to `eleventy.config.js`:

```js
const { EleventyHtmlBasePlugin } = require("@11ty/eleventy");

module.exports = function(eleventyConfig) {
	// ...

	eleventyConfig.addPlugin(EleventyHtmlBasePlugin);

	return {
		pathPrefix: "/tweetback/", // or "/twitter/", "/tweets/", etc.
	};
};
```

Thanks for the quick fix @zachleat 😊

**Comment by cooljeanius** [comment]
ah yeah I think this is what's causing cooljeanius/my_tweetback_archive#1; let me try the advice here...

**Comment by zachleat** [comment]
I believe this is a duplicate of fixed #13!

**Comment by cooljeanius** [comment]
> I believe this is a duplicate of fixed #13!

I think I might actually still be seeing it, though?
